File: lab_2/generator.cpp
#include <iostream>
#include <random>
#include <bitset>
#include <fstream>

void generate_cpp_sequence(int bits, const char* filename) {
    std::random_device rd;
    std::mt19937 gen(rd());
    std::uniform_int_distribution<> dis(0, 1);

    std::ofstream out(filename);
    for (int i = 0; i < bits; ++i) {
        out << dis(gen);
    }
    out.close();
}

int main() {
    generate_cpp_sequence(128, "cpp_random_bits.txt");
    return 0;
}

File: lab_2/generator.java
import java.io.FileWriter;
import java.io.IOException;
import java.util.Random;

public class RandomBitsGenerator {
    public static void generateJavaSequence(int bits, String filename) throws IOException {
        Random rand = new Random();
        FileWriter writer = new FileWriter(filename);

        for (int i = 0; i < bits; i++) {
            writer.write(rand.nextBoolean() ? "1" : "0");
        }

        writer.close();
    }

    public static void main(String[] args) throws IOException {
        generateJavaSequence(128, "java_random_bits.txt");
    }
}